src/Network/Pusher/Data.hs

```diff
@@ -1,4 +1,5 @@
 {-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE RecordWildCards #-}
```

Comment:
Given the fact that these are some pretty hefty records, I think pattern matching ended up looking rather ugly and went with the RecordWildCards extension. Removing it isn't an issue, but do you have a preferred convention when it comes to matching large records?

Are either of the following two options acceptable? Perhaps there is a cleaner way to do this that I'm just not thinking of?

```haskell
instance A.ToJSON Apns where
  toJSON apns@(Apns _ _ _ _ _ _) = A.object
    [ "alert" .= alert apns
    , "badge" .= badge apns
    , "sound" .= sound apns
    , "content-available" .= if contentAvailable apns then (1 :: Int) else 0
    , "category" .= category apns
    , "thread-id" .= threadId apns
    ]
```
My gut tells me this isn't a good way to do it, but it doesn't take too much line-width, and doesn't overshadow bindings.

```haskell
instance A.ToJSON Apns where
  toJSON (Apns alert badge sound contentAvailable category threadId) = A.object
    [ "alert" .= alert
    , "badge" .= badge
    , "sound" .= sound
    , "content-available" .= if contentAvailable then (1 :: Int) else 0
    , "category" .= category
    , "thread-id" .= threadId
    ]
```
This one doesn't look too bad, but this example only has 6 fields. `GcmNotification` currently has 12 and it would stretch way past 80 characters unless abbreviated names were used? Any recommendations for naming the captured values so as to not overshadow the record's automatically-generated "getter" functions?